### ISSUE TITLE
[RFE] Short name input format with SSSD for users from all domains when domain autodiscovery is used or when SSSD acts as an IPA client for server with IPA-AD trusts

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#3004]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/3001
- Created at 2016-04-27 17:09:46 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2017-03-31 18:13:42 as Fixed
- Assigned to [fidencio](https://pagure.io/user/fidencio)
---

Ticket was cloned from Red Hat Bugzilla (product *Red Hat Enterprise Linux 7*): [Bug 1330196](https://bugzilla.redhat.com/show_bug.cgi?id=1330196)

*Please note that this Bug is private and may not be accessible as it contains confidential Red Hat customer information.*

    1. Proposed title of this feature request
    
    Request for the ability to use short name input format with SSSD for users from
    all domains when domain autodiscovery is used.
    
    If each user exists only once in the whole forest, some users would like to use
    short name input for all users (for example for logins and commands which take
    a username argument), regardless as to which domain the user is joined to.
    Thous should be achieved with auto-discovery of domains in place, not by defining multiple domains

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-05-09 09:42:27](https://pagure.io/SSSD/sssd/issue/3001#240844)**

This is out of scope for 1.14, but since this feature is required quite often, I'm already marking it as a blocker for 1.15 (of course we would do another round of triage later).

blockedby:  => 
blocking:  => 
changelog:  => 
coverity:  => 
design:  => 
design_review:  => 0
feature_milestone:  => 
fedora_test_page:  => 
mark: no => 0
milestone: NEEDS_TRIAGE => SSSD 1.15 beta
priority: major => critical
review: True => 0
selected:  => 
testsupdated:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-09-30 23:35:20](https://pagure.io/SSSD/sssd/issue/3001#240845)**

When implementing this, note https://fedorahosted.org/sssd/ticket/3210

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 10:33:57](https://pagure.io/SSSD/sssd/issue/3001#240846)**

Fields changed

summary: [RFE] Short name input format with SSSD for users from all domains when domain autodiscovery is used => [RFE] Short name input format with SSSD for users from all domains when domain autodiscovery is used or when SSSD acts as an IPA client for server with IPA-AD trusts

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 10:34:32](https://pagure.io/SSSD/sssd/issue/3001#240847)**

Before tackling this ticket we need to:
    - switch the sysdb into using a unified name format everywhere (done)
    - switch the responders into using the cache_req request (in progress)
    - switch the cache_req code into querying all caches first before asking the Data Provider (in progress)

_comment0: Before tackling this ticket we need to:
 * switch the sysdb into using a unified name format everywhere (done)
 * switch the responders into using the cache_req request (in progress)
 * switch the cache_req code into querying all caches first before asking the Data Provider (not started) => 1486658602519824

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 10:35:49](https://pagure.io/SSSD/sssd/issue/3001#240848)**

Fields changed

blockedby:  => 3210

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-06 14:06:00](https://pagure.io/SSSD/sssd/issue/3001#240849)**

Fields changed

milestone: SSSD Future releases (no date set yet) => SSSD 1.15.1
owner: somebody => fidencio

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 15:03:07](https://pagure.io/SSSD/sssd/issue/3001#240850)**

**Metadata Update from @jhrozek**:
- Issue assigned to fidencio
- Issue marked as depending on: #3210
- Issue set to the milestone: SSSD 1.15.1

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-03 11:46:16](https://pagure.io/SSSD/sssd/issue/3001#429210)**

Since we are releasing 1.15.1 today, I'm moving all unfinished tickets from 1.15.1 to 1.15.2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-03 11:46:29](https://pagure.io/SSSD/sssd/issue/3001#429214)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue close_status updated to: None

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-03 11:47:54](https://pagure.io/SSSD/sssd/issue/3001#429224)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue set to the milestone: SSSD 1.15.2 (was: SSSD 1.15.1)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-03 12:10:57](https://pagure.io/SSSD/sssd/issue/3001#429241)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-14 12:37:08](https://pagure.io/SSSD/sssd/issue/3001#431121)**

First patches landed in master:
 * 606015a4f71d8ee809347188667d268f73110483
 * 8bb6680637ead03e24a38d15ec5265d11a920a1d
 * 9286d0d4143bcb4e36b91022107e307998122bbb
 * 7cd226414c7bcdd32f05416df64ebda3ac869bd7
 * 828fe7528fbe6971701d16aed87ba12dd91da55f

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-14 12:37:13](https://pagure.io/SSSD/sssd/issue/3001#431122)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-15 10:11:01](https://pagure.io/SSSD/sssd/issue/3001#431401)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue set to the milestone: SSSD 1.15.3 (was: SSSD 1.15.2)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-29 14:03:11](https://pagure.io/SSSD/sssd/issue/3001#434043)**

The rest of the patches landed as well:
 * 16385568547351b5d2c562f3081f35f3341f695b
 * 1e437af958f59a0b8bf2f751d3c2ea28365ac64d
 * 66c8e92eb5a4985bb7f64c349a53b08030a000cf
 * 34228050af1e25706f61ec9df648852284b61c2b
 * fb81f337b68c85471c3f5140850dccf549a2d0ac
 * 17ab121a6c69d74acf1d40f2bbcbe90d77bb6b8a
 * 3cbf0e7b63e8e6888917e9215bbdc5674c2fa852
 * 723d514f641e2b5a5cbfe1c6c7bdd2a6f3c5070e
 * 2e85b015d8dd231094a09eab69b86e8b6fcc8b2b
 * 5856a621ac5909ca96520ac5a809eb83fd46d8bc
 * a3442e4a268ad2172c89d58e6daa759eb4b39e7c
 * 46c99a59c8d6501aa3ad701c567fba577924b48b
 * dcc52d9c6411528bab815351d1e6145d211a4765
 * a63d74f65db2db7389cd373cb37adcdaaa2d56ea
 * e0e038218580166648ac24f23180f0f4c2769d99

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-29 14:03:19](https://pagure.io/SSSD/sssd/issue/3001#434044)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-31 18:09:04](https://pagure.io/SSSD/sssd/issue/3001#434530)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue **un**marked as depending on: #3210

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-31 18:13:49](https://pagure.io/SSSD/sssd/issue/3001#434539)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue close_status updated to: Fixed
- Issue status updated to: Closed (was: Open)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-04-26 21:17:57](https://pagure.io/SSSD/sssd/issue/3001#438521)**

Additonal patches:
 * b78febe4c579f86f8007a27599605d1eb9f97a62
 * 213048fd9a5e800deb74cb5b7f0eaf465945c640
 * f9bac02756aa05cc9c6ac07ae581dba67240c1a4
 * dae798231fc2c575f213785768bc24ed765ba243
 * ed518f61f1a5d4cf5d87eec492c158725a73d6a1
 * a3faad0e4dc1ca4473746c3822ecfc5aed876e6d

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-04-26 21:17:59](https://pagure.io/SSSD/sssd/issue/3001#438522)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
